app/src/main/java/com/example/secret/PostRecyclerAdapter.java

```diff
@@ -0,0 +1,113 @@
+package com.example.secret;
+
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.CheckBox;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import androidx.annotation.NonNull;
+import androidx.recyclerview.widget.RecyclerView;
+
+import com.example.secret.model.Comment;
+import com.example.secret.model.Post;
+import com.squareup.picasso.Picasso;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+
+class PostViewHolder extends RecyclerView.ViewHolder {
+    List<Post> posts;
+    Map<String, List<Comment>> postsLatestComments;
+    ImageView backgroundImage;
+    TextView contentTv;
+    TextView comment1Tv;
+    TextView comment2Tv;
+
+    public PostViewHolder(@NonNull View itemView, PostRecyclerAdapter.OnItemClickListener listener, List<Post> posts, Map<String, List<Comment>> postsLatestComments) {
```

Comment:
why not pass here only the related post and its comments?

Reply:
Its the same as Eliav did